include/manif/impl/sgal3/SGal3_base.h

```diff
@@ -0,0 +1,484 @@
+#ifndef _MANIF_MANIF_SGAL3_BASE_H_
+#define _MANIF_MANIF_SGAL3_BASE_H_
+
+#include "manif/impl/sgal3/SGal3_properties.h"
+#include "manif/impl/lie_group_base.h"
+#include "manif/impl/so3/SO3_map.h"
+#include "manif/impl/se3/SE3_map.h"
+
+namespace manif {
+
+//
+// LieGroup
+//
+
+/**
+ * @brief The base class of the SGal3 group.
+ * @note See "All About the Galilean Group SGal(3)" J. Kelly.
+ * https://arxiv.org/abs/2312.07555
+ *
+ */
+template <typename _Derived>
+struct SGal3Base : LieGroupBase<_Derived> {
+private:
+
+  using Base = LieGroupBase<_Derived>;
+  using Type = SGal3Base<_Derived>;
+
+public:
+
+  MANIF_GROUP_TYPEDEF
+  MANIF_INHERIT_GROUP_AUTO_API
+  MANIF_INHERIT_GROUP_OPERATOR
+
+  using Base::coeffs;
+
+  using Rotation       = typename internal::traits<_Derived>::Rotation;
+  using Translation    = typename internal::traits<_Derived>::Translation;
+  using LinearVelocity = typename internal::traits<_Derived>::LinearVelocity;
+  using Time           = Scalar;
+  using Transformation = Eigen::Matrix<Scalar, 5, 5>;
+  using Isometry       = Eigen::Matrix<Scalar, 5, 5>; /**< Double direct spatial isometry*/
+  using QuaternionDataType = Eigen::Quaternion<Scalar>;
+
+  // LieGroup common API
+
+protected:
+
+  using Base::derived;
+
+  MANIF_DEFAULT_CONSTRUCTOR(SGal3Base)
+
+public:
+
+  MANIF_GROUP_ML_ASSIGN_OP(SGal3Base)
+
+  /**
+   * @brief Get the inverse.
+   * @param[out] -optional- J_minv_m Jacobian of the inverse wrt this.
+   */
+  LieGroup inverse(OptJacobianRef J_minv_m = {}) const;
+
+  /**
+   * @brief Get the SGal3 corresponding Lie algebra element in vector form.
+   * @param[out] -optional- J_t_m Jacobian of the tangent wrt to this.
+   * @return The SGal3 tangent of this.
+   * @note This is the log() map in vector form.
+   * @see SGal3Tangent.
+   */
+  Tangent log(OptJacobianRef J_t_m = {}) const;
+
+  /**
+   * @brief This function is deprecated.
+   * Please considere using
+   * @ref log instead.
+   */
+  MANIF_DEPRECATED
+  Tangent lift(OptJacobianRef J_t_m = {}) const;
+
+  /**
+   * @brief Composition of this and another SGal3 element.
+   * @param[in] m Another SGal3 element.
+   * @param[out] -optional- J_mc_ma Jacobian of the composition wrt this.
+   * @param[out] -optional- J_mc_mb Jacobian of the composition wrt m.
+   * @return The composition of 'this . m'.
+   */
+  template <typename _DerivedOther>
+  LieGroup compose(
+    const LieGroupBase<_DerivedOther>& m,
+    OptJacobianRef J_mc_ma = {},
+    OptJacobianRef J_mc_mb = {}
+  ) const;
+
+  /**
+   * @brief Get the action
+   * @param[in] v A 3D point.
+   * @param[out] -optional- J_vout_m The Jacobian of the new object wrt this.
+   * @param[out] -optional- J_vout_v The Jacobian of the new object wrt input object.
+   */
+  template <typename _EigenDerived>
+  Eigen::Matrix<Scalar, 3, 1>
+  act(
```

Comment:
I was thinking, regarding act() in all complex groups. Maybe we could have special actions. In case of SGal3, we would have:

act: T+Rp
boost: see paper
event-transform: see paper

and other possible actions only available for this group.

In SE3 we have a similar problem, in transforming points (T+R*p) and transforming vetors (R*v)

I'll add this comment as an issue in manif, for consideration.

Reply:
Disregard this comment in this review.